### DISCUSSION TITLE
Merge project with ESP32-HomeKey

### DISCUSSION THREAD

**Discussion by konradhuebner** [omersiar/esp-rfid#653] [Ideas]
Hey @matjack1 as you seem to be the current soul of this project: I played with esp-rfid, and even the hardware boards one can get, and I also checked out this one: https://github.com/rednblkx/HomeKey-ESP32 which supports Apple HomeKey, but also reading standard RFID tags. However, that project is also limited to one NFC reader right now. I was wondering if it doesn't make sense to build a merge of the two, and join forces. Having a local user / access control, and direct relay access feels a good thing to have. A user management is great as well. And all that with Apple HomeKey is really nice. Thoughts? Will limit to ESP32 though, but that one got so cheap that it shouldn't do any harm. Coolest ever would be a v2 of the hardware and finding some manufacturer for that.

**Comment by matjack1** [comment]
hey @konradhuebner thank you for the idea! I'm not an iPhone user myself, so I didn't know about Homekit, but surely looks very useful to add to esp-rfid :)

Merging the two projects looks pretty hard at the moment, really.

I've seen a port of esp-rfid for ESP32: https://github.com/pvtex/esp32-rfid which I haven't tried yet, but it could be a good place to start if we want to evolve this project to ESP32, mainly to get a bit more stability when dealing with a lot of data.

Currently though, I'll try to finish the v2 with what I have now in terms of features. I need to test all the hardware supported and if everything works ship a v2. After that I'm not sure if it's worth porting to ESP32 or doing something totally different.

Surely Homekit seems very interesting, but like Homeassistant, they are systems that don't really leverage what makes esp-rfid particular, which is the ability to run completely standalone, without needing anything else to work.

**Reply by konradhuebner** [reply]
@matjack1  thanks for the answer. I also do like the autonomy of esp-rfid, and I think this can be achieved when merging as well. The HomeKey feature would just be an addition for people who like it. There is also another project doing a fingerprint recognition based on ESP32. If you marry the three projects, it would be the ultimate access control system. Having a local user management and a local relais to execute actions is for sure something I would also prefer.

If you need some support in testing the v2, let me know. I do have some hardware here and could give it a try.

**Reply by matjack1** [reply]
I see, yes, it would mean to upgrade esp-rfid to ESP32 first, then it would enable all those sensors/projects to be added, it makes sense.

If you want to test v2, please do already! It's missing really a couple of things so if you can test it already it will be very useful. Thank you!!
